app/Main.hs

```diff
@@ -120,27 +129,65 @@ data Message
 
 main :: IO ()
 main = do
-  Opts{..} <- execParser parserInfo
+  opts@Opts{..} <- execParser parserInfo
+  case optsCommand of
+    Deploy cliReleaseFormat cliKeepReleases cliKeepOneFailed ->
+      runHapCmd opts $ \hapConfig@C.Config{..} executionMode ->
+        Hap.deploy
+          hapConfig
+          (fromMaybeReleaseFormat cliReleaseFormat configReleaseFormat)
+          (fromMaybeKeepReleases cliKeepReleases configKeepReleases)
+          (cliKeepOneFailed || configKeepOneFailed)
+          executionMode
+    Rollback n ->
+      runHapCmd opts $ \C.Config{..} _ ->
+        Hap.rollback configTargetSystem configDeployPath n configRestartCommand
+    Maintenance Enable ->
+      runHapCmd opts $ \C.Config{..} _ ->
+        Hap.writeMaintenanceFile configDeployPath configMaintenanceDirectory configMaintenanceFileName
+    Maintenance _ ->
+      runHapCmd opts $ \C.Config{..} _ ->
+        Hap.deleteMaintenanceFile configDeployPath configMaintenanceDirectory configMaintenanceFileName
+    InitConfig -> do
+      alreadyExisting <- doesFileExist "hap.yml"
+      when alreadyExisting $ do
+        hPutStrLn stderr "'hap.yml' already exists"
+        exitFailure
+      putStrLn "Creating 'hap.yml'"
+      defaults <- defaultInitTemplateConfig
+      let prompt :: Read a => T.Text -> a -> IO a
+          prompt title d = do
+            T.putStrLn $ title <> "?: "
+            x <- getLine
+            return $
+              if null x
+                then d
+                else read x
+          prompt' :: Read a => T.Text -> (InitTemplateConfig -> T.Text) -> (InitTemplateConfig -> a) -> IO a
+          prompt' title f fd = prompt (title <> " (default: " <> f defaults <> ")") (fd defaults)
+
+      let yesNo :: a -> a -> T.Text -> a
+          yesNo t f x = if x == "y" then t else f
+
+      config <-
+        InitTemplateConfig
+          <$> prompt' "repo" repo repo
+          <*> prompt' "revision" revision revision
+          <*> prompt' "host" host host
+          <*> prompt' "port" (T.pack . show . port) port
+          <*> return (buildScript defaults)
+          <*> fmap (yesNo (restartCommand defaults) Nothing) (prompt' "Include restart command" (const "Y/n") (const "y"))
+
+      Yaml.encodeFile "hap.yml" config
```

Comment:
@blackheaven Can you move these lines inside the `System.Hapistrano` and associate it with a function like we do for `deploy`, `rollbar`, `deleteMaintenanceFile`, etc.? I think it's important to keep consistency. Additionally, we could add some tests for that new function.

Reply:
I moved it.

However, I'm not sure which tests I can write, as it is tied to IO (`readLine`, `putStrLn`, accessing to the file system).

Reply:
@CristhianMotoche do you have any tests you'd like to have?

Reply:
Hello @blackheaven Sorry for the delay. I think we could add a single test making sure the execution of that new command create the `hap.yml` file. Just making sure the [file does exist](https://hackage.haskell.org/package/directory-1.3.8.1/docs/System-Directory.html#v:doesFileExist).

Reply:
Is it enough, or should I add more?

Reply:
Hi @blackheaven I think that's good enough! Thank you!